pkg/types/validation/installconfig.go

```diff
@@ -103,21 +108,28 @@ func validateClusterNetwork(cn *netopv1.ClusterNetwork, fldPath *field.Path, ser
 	return allErrs
 }
 
-func validateMachinePools(pools []types.MachinePool, fldPath *field.Path, platform string) field.ErrorList {
+func validateControlPlane(pool *types.MachinePool, fldPath *field.Path, platform string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if pool.Name != "master" {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), pool.Name, []string{"master"}))
+	}
+	allErrs = append(allErrs, ValidateMachinePool(pool, fldPath, platform)...)
+	return allErrs
+}
+
+func validateCompute(pools []types.MachinePool, fldPath *field.Path, platform string) field.ErrorList {
 	allErrs := field.ErrorList{}
 	poolNames := map[string]bool{}
 	for i, p := range pools {
```

Comment:
We should also check that `len(pools) > 0`, to work around [the router issue][1].

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1671136#c1

Reply:
Yes. I will change that.